hbase_sql/src/main/java/com/splicemachine/olap/AsyncOlapNIOLayer.java

```diff
@@ -440,16 +442,24 @@ public void operationComplete(Future<Void> future) throws Exception{
         @Override
         protected void channelRead0(ChannelHandlerContext ctx,OlapMessage.Response olapResult) throws Exception{
             OlapResult or=parseFromResponse(olapResult);
+            if (LOG.isTraceEnabled()) {
+                LOG.trace("Received " + or);
+            }
             //TODO -sf- deal with a OlapServer failover here (i.e. a move to NOT_SUBMITTED from any other state
             if(or instanceof SubmittedResult) {
                 future.tickTimeNanos = TimeUnit.MILLISECONDS.toNanos(((SubmittedResult) or).getTickTime());
                 future.lastStatus = System.currentTimeMillis();
             } else if(future.submitted && !future.isDone() && or instanceof NotSubmittedResult) {
-                // The job is no longer submitted, assume aborted
+                // Server says the job is no longer submitted, give it a couple of tries in case messages are out of order
```

Comment:
We request the status a fixed time after the previous request, not after the previous answer. Furthermore we use a different channel for each request, so I'm not sure we can guarantee in order processing of the requests.